packages/SidePanel/src/hooks/useOffsetScroll.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+const useOffsetScroll = offsetY => {
+  const [offsetScroll, setOffsetScroll] = React.useState(offsetY);
+
+  const handleScroll = () => {
+    const top = window.pageYOffset || document.documentElement.scrollTop;
+    if (top <= offsetY) {
+      setOffsetScroll(offsetY - top);
+    } else {
+      setOffsetScroll(0);
+    }
+  };
+
+  React.useLayoutEffect(() => {
```

Comment:
During the advance hooks workshop, kent dodds demonstrated an [example](https://github.com/kentcdodds/advanced-react-hooks/blob/master/src/exercises/05.js#L8) pretty similar to this, he advice was if the effect will affect the visual element while interacting with the browser API, use Layout otherwise useEffect. 

That's the reason I'm using *useLayout*, also he even demonstrated that because react is really fast you can't perceive really a difference between useEffect vs useLayoutEffect in this case, that's the reason why he added a sleep function in [this line](https://github.com/kentcdodds/advanced-react-hooks/blob/master/src/exercises/05.js#L49) so the difference is more perceivable.

Reply:
That example is different—it really does "affect the visual element" directly, while yours only sets a listener which has no (immediate) visual implications. In that case, `useEffect` is recommended since it allows React to schedule it better and reduce the workload in the commit phase.

Reply:
`useEffect`
![Screen Shot 2019-06-26 at 2 29 15 PM](https://user-images.githubusercontent.com/19418590/60216583-4cb74c80-981f-11e9-8ed5-281fa8888e88.png)

`useLayoutEffect`
![Screen Shot 2019-06-26 at 2 29 44 PM](https://user-images.githubusercontent.com/19418590/60216584-4cb74c80-981f-11e9-851b-efa04300d25e.png)

see how the Sidepanel is not a `0px` when scrolling? and in the second example when scrolling is actually at `0px`. 

Changing to useEffect will never update cypress on the scrolling event, useLayoutEffect occurs before useEffect in the cycle and is synchronous so will update the element and cypress can observe the changes.

Reply:
I'd say that's probably a cypress issue then. But not going to block on this for sure